### PR TITLE
Implementing Health Item

### PR DESCRIPTION
**Pull request by AleksKors**
### What has been done?
- Implemented HealthItem which  extends from CommonItem and implements IHealingEntity.
- Implemented logic in players collide method
- Implemented tests

### CODE REVIEW

File: HealthItem/src/main/java/module-info.java [hunk 1]
```diff
@@ -1,2 +1,9 @@
+import com.github.sef24sp4.common.services.IGamePluginService;
+import com.github.sef24sp4.healthitem.HealthItemPlugin;
+
 module HealthItem {
```
Comment (MidnightRocket) [inline HealthItem/src/main/java/module-info.java:4]: You need to add a provides for `HealthItemProvider`

File: HealthItem/src/test/java/com/github/sef24sp4/healthitem/HealthItemTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class HealthItemTest {
+	@Mock
+	private IEntityManager mockEntityManager = mock(IEntityManager.class);
+
+	@Mock
+	private ICollidableEntity mockEntity = mock(ICollidableEntity.class);
```
Comment (MidnightRocket) [inline HealthItem/src/test/java/com/github/sef24sp4/healthitem/HealthItemTest.java:18]: When using the `@Mock` annotation you should omit the instantiation, and instead use the `@ExtendsWith(MockitoExtension)` annotation. [Example](https://github.com/SE-F24-SP4/semesterproject-4/blob/a44129194675e5a05197e91b019856dd4f2e58f3/CollisionSystem/src/test/java/com/github/sef24sp4/collisionsystem/CollisionSystemTest.java#L20-L25)

File: HealthItem/src/test/java/com/github/sef24sp4/healthitem/HealthItemTest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+	@Mock
+	private HealthItem healthItem = new HealthItem();
+
+	@Mock
+	private HealthItemProvider healthItemProvider = new HealthItemProvider();
```
Comment (MidnightRocket) [inline HealthItem/src/test/java/com/github/sef24sp4/healthitem/HealthItemTest.java:24]: You should not use the `@Mock` annotation on non mocked fields.

File: HealthItem/src/test/java/com/github/sef24sp4/healthitem/HealthItemTest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+		verify(this.mockEntityManager).removeEntity(this.healthItem);
+	}
+
+	@Test
+	void collideNotPlayer() {
+	    //Setup
+	    this.mockEntityManager.addEntity(this.healthItem);
+		when(this.mockEntity.getType()).thenReturn(GameElementType.ENEMY);
+		//Method call
+		this.healthItem.collide(this.mockEntityManager, this.mockEntity);
```
Comment (MidnightRocket) [inline HealthItem/src/test/java/com/github/sef24sp4/healthitem/HealthItemTest.java:43]: Please fix indentation

File: HealthItem/src/test/java/com/github/sef24sp4/healthitem/HealthItemTest.java
Comment (MidnightRocket) [inline HealthItem/src/test/java/com/github/sef24sp4/healthitem/HealthItemTest.java:1]: Great use of the Mockito framework. There are still some small improvements to be made.

**Review by MidnightRocket — CHANGES_REQUESTED** [review]
Great work and good tests using the Mockito framework. There are stille som improvements to made though.

File: HealthItem/src/test/java/com/github/sef24sp4/healthitem/HealthItemTest.java (hunk 4 above)
Comment (AleksKors) [reply HealthItem/src/test/java/com/github/sef24sp4/healthitem/HealthItemTest.java:43]: Not quiet sure whats wrong with the marked lines?

Comment (MidnightRocket) [reply HealthItem/src/test/java/com/github/sef24sp4/healthitem/HealthItemTest.java:43]: These lines are indented one level too deep

Comment (AleksKors) [reply HealthItem/src/test/java/com/github/sef24sp4/healthitem/HealthItemTest.java:43]: I can see the indentation on github, but not when accessing the file from a file editor lol? Thought it mby was a problem with IntelliJ, but I even tried with notepad :/

Comment (MidnightRocket) [reply HealthItem/src/test/java/com/github/sef24sp4/healthitem/HealthItemTest.java:44]: Try in Intellij to reformat the whole file.

**Review by MidnightRocket — APPROVED** [review]
Looks good now 👍

**Pull request merged**